ffi/src/lib.rs

```diff
@@ -1755,3 +1776,275 @@ pub(crate) unsafe fn get_pipeline_options(
+/// - `traces`: Optional configuration for exporting trace data. If `None`, trace data will not be exported.
+/// - `metrics`: Optional configuration for exporting metrics data. If `None`, metrics data will not be exported.
+/// - `flush_interval_ms`: Optional interval in milliseconds between consecutive exports of telemetry data. If `None`, a default value will be used.
+///
+/// At least one of traces or metrics must be provided.
+#[repr(C)]
+#[derive(Clone)]
+pub struct OpenTelemetryConfig {
+    /// Whether traces configuration is present
+    pub has_traces: bool,
+    /// Configuration for exporting trace data. Only valid if has_traces is true.
+    pub traces: OpenTelemetryTracesConfig,
+    /// Whether metrics configuration is present
+    pub has_metrics: bool,
+    /// Configuration for exporting metrics data. Only valid if has_metrics is true.
+    pub metrics: OpenTelemetryMetricsConfig,
+    /// Whether flush interval is specified
+    pub has_flush_interval_ms: bool,
+    /// Interval in milliseconds between consecutive exports of telemetry data. Only valid if has_flush_interval_ms is true.
+    pub flush_interval_ms: i64,
+}
+
+/// Configuration for exporting OpenTelemetry traces.
+///
+/// - `endpoint`: The endpoint to which trace data will be exported. Expected format:
+///   - For gRPC: `grpc://host:port`
+///   - For HTTP: `http://host:port` or `https://host:port`
+///   - For file exporter: `file:///absolute/path/to/folder/file.json`
+/// - `has_sample_percentage`: Whether sample percentage is specified
+/// - `sample_percentage`: The percentage of requests to sample and create a span for, used to measure command duration. Only valid if has_sample_percentage is true.
+#[repr(C)]
+#[derive(Clone)]
+pub struct OpenTelemetryTracesConfig {
+    /// The endpoint to which trace data will be exported.
+    pub endpoint: *const c_char,
+    /// Whether sample percentage is specified
+    pub has_sample_percentage: bool,
+    /// The percentage of requests to sample and create a span for, used to measure command duration. Only valid if has_sample_percentage is true.
+    pub sample_percentage: u32,
+}
+
+/// Configuration for exporting OpenTelemetry metrics.
+///
+/// - `endpoint`: The endpoint to which metrics data will be exported. Expected format:
+///   - For gRPC: `grpc://host:port`
+///   - For HTTP: `http://host:port` or `https://host:port`
+///   - For file exporter: `file:///absolute/path/to/folder/file.json`
+#[repr(C)]
+#[derive(Clone)]
+pub struct OpenTelemetryMetricsConfig {
+    /// The endpoint to which metrics data will be exported.
+    pub endpoint: *const c_char,
+}
+
+/// Initializes OpenTelemetry with the given configuration.
+///
+/// # Safety
+/// * `open_telemetry_config` and its underlying traces and metrics pointers must be valid until the function returns.
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn init_open_telemetry(
+    open_telemetry_config: OpenTelemetryConfig,
```

Comment:
maybe use ptr to avoid struct copying on call?
```suggestion
    open_telemetry_config: *const OpenTelemetryConfig,
```